doc/feature/validation.md

```diff
@@ -0,0 +1,80 @@
+====================================================
+
+Since LPF is a physically based computation we need a way to validate the
+results against the electromagnetic theory. We also should evaluate the impact
+of the cuttof distance on the errors between our computations and the ideal
+case i.e. infinite cutoff distance. Another step which is beyong pure algorithm
+validation would be to compute the errors between full compartments reports and
+soma only reports simulations.
+
+## Requirements
+
+* A simple setup( 1-5 point sources ) simulation should be compared to pure
+pencil/paper computation.
+* We need to create several graphs showing the errors decrease with the increase 
+of the cutoff distance.
+* We need to compute errors betweens full compartments reports and soma only 
+reports simulations.
+* We need to create a document to show the results to scientists.
+* Create tests.
+
+## Implementation
+
+To achieve all these validations, we need to have a volume exporter and a
+volume comparator.
+
+### Volume exporter
+
+The volume exporter must be very precise so one can easily map a voxel position
+into a real physical position. Any corner/border case errors must be avoided. 
+This is critical for example in the first requirement point above, one need to 
+compute by hand a field intensity value for a physical position, this physical 
+position should perfecty (in the range of floating point precision) match the 
+center of a voxel. Otherwise validation will be broken.
+
+Maybe it is possible to add some meta-data at the end of the file. Usefull 
+meta-data can be volume resolution(voxels/um), position of the circuit bounding
+box, origin, etc... The circuit bounding box can be very usefull to compare 
+volumes of different sizes especially knowing the fact that in our current 
+implementation different cuttof distance will generate different size volumes. 
+This point is open for discussion.
+
+### Volume comparator
+
+The volume comparator must take 2 volumes as input and ouput some difference 
+measurements. An obvious one is the mean-square error. The maximum square error
+can be also interesting. We can use some command line parameters to choose which
+measurement we want or compute always everything.
+
+Because most of the time volumes will have different sizes the comparator should
+have some additional information about the location of the circuit bounding box.
+We must be totally sure that only voxels with the exact same physical location 
+(in the range of floating point precision) are compared toghether. This can be 
+achieved either by providing this information in some command line parameters or
+by embedding it in the .raw volume file as discussed before.
+
+The comparator will take the smallest volume and position it in the bigger 
+volume so the physical positions of the voxels will match. After that it will 
+iterate over all voxels in the smaller volume and compute the errors.
```

Comment:
I guess this would be a new tool. Maybe one of the filters already available in ITK can help you here, something like https://itk.org/Wiki/ITK/Examples/ImageProcessing/SquaredDifferenceImageFilter might be worth a try.